InfiniteRecharge/src/main/java/frc/robot/utils/dashboard/DashboardHandler.java

```diff
@@ -0,0 +1,252 @@
+        Annotation[] annotations = clazz.getAnnotations();
+        for (Annotation annotation : annotations) {
+            if (annotation instanceof Dashboard) {
+                return ((Dashboard) annotation).populateOnCreate();
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Build a dashboard key for a method name
+     *
+     * @param methodName The name of the method to get a key for
+     * @return
+     */
+    String keyForMethodName(String methodName) {
+        String entryKey = StringUtils.capitalize(StringUtils.join(StringUtils.splitByCharacterTypeCamelCase(methodName), ' '));
+        if (entryKey.startsWith("Put ") || entryKey.startsWith("Get ") || entryKey.startsWith("Set ")) {
+            return name + "/" + entryKey.substring(4);
+        } else {
+            return name + "/" + entryKey;
+        }
+    }
+
+    /**
+     * Invoke any methods on this interface as SmartDashboard methods
+     *
+     * @param proxy
+     * @param method
+     * @param args
+     * @return
+     * @throws Throwable
+     */
+    @Override
+    public Object invoke(Object proxy, Method method, Object[] args) throws Throwable {
+
+        // make sure toString for the proxy works in the debugger.
+        if (method.getName().equals("toString")) {
+            return proxy.getClass().getName();
+        }
+
+        String key = keyForMethodName(method.getName());
+        DashboardEntry entry = entries.get(key);
+        if (entry == null) {
+            System.err.println("Attempting to invoke unknown method in Dashboard interface: " + method.getName());
+            return null;
+        }
+
+        if (entry.setter) {
+            switch (entry.type) {
+                case Number:
+                    SmartDashboard.putNumber(entry.dashboardKey, (Double) args[0]);
+                    break;
+                case String:
+                    SmartDashboard.putString(entry.dashboardKey, (String) args[0]);
+                    break;
+                case Boolean:
+                    SmartDashboard.putBoolean(entry.dashboardKey, (Boolean) args[0]);
+                    break;
+            }
+            // for setters, we return null
+            return null;
+        } else {
+            switch (entry.type) {
+
+                case Number:
+                    return getCache.get(entry.dashboardKey, () -> SmartDashboard.getNumber(entry.dashboardKey, (Double) entry.defaultValue));
+                case String:
+                    return getCache.get(entry.dashboardKey, () -> SmartDashboard.getString(entry.dashboardKey, (String) entry.defaultValue));
+                case Boolean:
+                    return getCache.get(entry.dashboardKey, () -> SmartDashboard.getBoolean(entry.dashboardKey, (Boolean) entry.defaultValue));
+            }
+        }
+
+
+        throw new IllegalArgumentException("Failed to query SmartDashboard for this value: " + method.getName() + " " + method.getReturnType());
+    }
+}
```

Comment:
add two more lines of code here and put the comment

Reply:
I do not get it...

Reply:
Oh.